src/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/ExternalReferencesTable.cs

```diff
@@ -95,8 +91,9 @@ unsafe public IntPtr GetIntPtrFromIndex(uint index)
                 throw new BadImageFormatException();
 
             // TODO: indirection through IAT
+            int* elementsTable = (int*)_elements;
 
-            return ((TableElement*)_elements)[index];
+            return (IntPtr)((byte*)&elementsTable[index + 1] + elementsTable[index]);
```

Comment:
With `IMAGE_REL_BASED_REL32`, this can be something like:

```
int* pRelPtr32 = ((int*)_elements[index]);
return (IntPtr)((byte*)pRelPtr32 + *pRelPtr32);
```